File: 0002-add-two-numbers/0002-add-two-numbers.js
  function ListNode(val, next) {
      this.val = (val===undefined ? 0 : val)
      this.next = (next===undefined ? null : next)
  }
  /*
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
    let over = 0;
    const recursive = () =>{
        if(l1 === null && l2 === null){
            if(over === 1){
                return new ListNode(1, null);
            }
            return null;
        }

        let sum = over;
        if(l1 !== null){
            sum += l1.val; 
            l1 = l1.next;
        }
        if(l2 !== null){
            sum += l2.val;
            l2 = l2.next;
        }
        if(sum >=10){
            over = 1;
        }else{
            over = 0; 
        }

        return new ListNode(sum % 10, recursive())
    }
    return recursive();
};

File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if(l1 == null && l2 == null) return null;
        if(l1 == null) l1 = new ListNode(0);
        if(l2 == null) l2 = new ListNode(0);

        var value = l1.val + l2.val;
        if(value >= 10) {
            if(l1.next == null) l1.next = new ListNode(0);
            l1.next.val += 1;
            return new ListNode(value - 10, addTwoNumbers(
                l1.next,
                l2.next
            ));
        } 
        else return new ListNode(value, addTwoNumbers(l1.next, l2.next));
    }
}